proto.go

```diff
@@ -0,0 +1,26 @@
+package matchers
+
+import (
+	"github.com/golang/mock/gomock"
+	"google.golang.org/protobuf/proto"
+)
+
+type ProtoMatcher struct {
+	data proto.Message
+}
+
+func Proto(x proto.Message) gomock.Matcher {
+	return &ProtoMatcher{data: x}
+}
+
+func (p *ProtoMatcher) Matches(x interface{}) bool {
+	pbx, ok := x.(proto.Message)
+	if !ok {
+		return false
+	}
+	return proto.Equal(p.data, pbx)
+}
+
+func (p *ProtoMatcher) String() string {
+	return "is matching"
```

Comment:
This is the most sophisticated `String()` function I could come up with... not much useful info in `proto.Message`

Reply:
This string is supposed to explain what is in `p.data` - perhaps a JSON serialization? Something like

```go
data, err := json.Marshal(p.data)
if err != nil {
  return "<unknown>"
}
return data
```